### PR TITLE
Add the http protocol to urls that lack it

### PR DESCRIPTION
**Pull request by r00k**
If a user enters google.com, let's add the http:// for her.

### CODE REVIEW

File: app/models/url_normalizer.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class UrlNormalizer
+
```
Comment (seanpdoyle) [inline app/models/url_normalizer.rb]: ✂️

File: spec/models/url_normalizer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+require "rails_helper"
+
+describe UrlNormalizer do
+  describe "#url" do
+    context "when the url lacks a protocal" do
```
Comment (seanpdoyle) [inline spec/models/url_normalizer_spec.rb]: `protocol`

File: app/models/url_normalizer.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  def initialize(url)
+    @url = url
+  end
+
+  def url
+    if @url.include?("http")
+      @url
+    else
+      "http://" + @url
```
Comment (seanpdoyle) [inline app/models/url_normalizer.rb]: What do you think about checking for the presence of this string (including the `://`)?

File: app/controllers/analyses_controller.rb [hunk 4]
```diff
@@ -1,7 +1,19 @@
...
+  def form_html
+    FormFinder.new(page_body).find
+  end
+
+  def page_body
+    HTTParty.get(url)
+  end
+
+  def url
+    UrlNormalizer.new(params[:analysis][:url]).url
```
Comment (seanpdoyle) [inline app/controllers/analyses_controller.rb]: What do you think about constructing the `Analysis` like a conventional Rails model?

That way, we could move most of this logic out of the controller and into the model:

```rb
def create
  @analysis = Analysis.new(analysis_params)
end

def analysis_params
  params.require(:analysis).permit(:url)
end
```

File: spec/models/url_normalizer_spec.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      it "adds http" do
+        url = "example.com"
+
+        result = UrlNormalizer.new(url).url
+
+        expect(result).to eq("http://example.com")
+      end
+    end
+
+    context "when the url includes a protocal" do
```
Comment (seanpdoyle) [inline spec/models/url_normalizer_spec.rb]: `protocol`

File: app/models/url_normalizer.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+class UrlNormalizer
```
Comment (seanpdoyle) [inline app/models/url_normalizer.rb]: Could we outsource this behavior to a gem?

File: app/models/url_normalizer.rb (hunk 3 above)
Comment (r00k) [inline app/models/url_normalizer.rb]: That breaks if the user entered `https://`. I'll add a test case for that.

File: app/models/url_normalizer.rb (hunk 6 above)
Comment (r00k) [inline app/models/url_normalizer.rb]: We googled around for a few minutes but couldn't find an obvious solution.

File: app/controllers/analyses_controller.rb (hunk 4 above)
Comment (r00k) [inline app/controllers/analyses_controller.rb]: Done!

**Comment by seanpdoyle** [comment]
Nice work!

**Pull request merged**